packages/material-ui-styles/src/styled/styled.d.ts

```diff
@@ -6,6 +6,12 @@ import {
 } from '@material-ui/styles/withStyles';
 import * as React from 'react';
 
+/**
+ * If props is any, returns empty object, otherwise T
+ * @internal
+ */
+export type GetProps<T> = 0 extends 1 & T ? {} : T;
```

Comment:
nit: name it `CoerceEmptyInterface` and use `IsAny` from `makeStyles.d.ts`.

Also: The blanket export was because I didn't know how to disable it. Even without the `export` keyword the type would've been importable. Turns out that you can disable export by default by adding `export {}` to the file. I would prefer that for files that use types that are only used internally.

Reply:
Should we move them to `@material-ui/types` instead?
`CoerceEmptyInterface` will be used in core later, so it should probably stay exported

Reply:
> Should we move them to @material-ui/types instead?

Yeah that sounds perfect. 

There are plenty of other types that can be moved there. E.g. most components also touch the index.d.ts file because it includes some utility types. Moving those into a separate package could help TS compilation times as well

> CoerceEmptyInterface will be used in core later, so it should probably stay exported

Alright.

Reply:
Done